File: app/src/main/java/vergecurrency/vergewallet/view/ui/activity/settings/ShowPaperkeyActivity.java
package vergecurrency.vergewallet.view.ui.activity.settings;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;

import vergecurrency.vergewallet.view.base.BaseActivity;
import vergecurrency.vergewallet.R;
import vergecurrency.vergewallet.viewmodel.ShowPaperkeyViewModel;

public class ShowPaperkeyActivity extends BaseActivity {

	private ShowPaperkeyViewModel mViewModel;
	private TextView seedTextView;
	private ImageView closeButton;


	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		mViewModel = ViewModelProviders.of(this).get(ShowPaperkeyViewModel.class);
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_paperkey_shown);
		initComponents();
	}

	//TODO : require passphrase before showing seed, create function for toggling between
	// qr and text views

	private void initComponents() {
		seedTextView = findViewById(R.id.paperkey_shown_pk);
		seedTextView.setText(mViewModel.getSeedAsText());
		closeButton = findViewById(R.id.paperkey_shown_close_button);
		closeButton.setOnClickListener(onCloseClickListener());
	}

	View.OnClickListener onCloseClickListener() {
		return v -> {
			//finish();
			onBackPressed();
		};
	}

}

File: app/src/main/kotlin/vergecurrency/vergewallet/view/ui/activity/settings/ShowPaperkeyActivity.kt
package vergecurrency.vergewallet.view.ui.activity.settings

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.lifecycle.ViewModelProviders
import vergecurrency.vergewallet.R
import vergecurrency.vergewallet.view.base.BaseActivity
import vergecurrency.vergewallet.viewmodel.ShowPaperkeyViewModel

class ShowPaperkeyActivity : BaseActivity() {

    private var mViewModel: ShowPaperkeyViewModel? = null
    private var seedTextView: TextView? = null
    private var closeButton: ImageView? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        mViewModel = ViewModelProviders.of(this).get(ShowPaperkeyViewModel::class.java)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_paperkey_shown)
        initComponents()
    }


    private fun initComponents() {
        seedTextView = findViewById(R.id.paperkey_shown_pk)
        seedTextView!!.text = mViewModel!!.seedAsText
        closeButton = findViewById(R.id.paperkey_shown_close_button)
        closeButton!!.setOnClickListener(onCloseClickListener())
    }

    internal fun onCloseClickListener(): View.OnClickListener {
        return View.OnClickListener { finish() }
    }

}
